### PR TITLE
Stats: Replace the use of the PostStatsStore by the redux approach (query + selector)

### PR DESCRIPTION
**Pull request by youknowriad**
I replaced the use of the PostStatsStore by the redux approach (query + selector)
I also dropped the store entirely (not used anymore) and added a heartbeat prop the QueryPostStat component to refresh the results each n milliseconds

**Testing Instructions**

No functional changes.

 * Navigate to my sites blog posts page
 * Check that the posts' view count is displayed correctly

cc @timmyc @mdawaffe

### CODE REVIEW

**Comment by youknowriad** [comment]
@timmyc In my opinion the heartbeat (polling) is not required. I kept this iso functional for now but I can drop the polling here.

**Comment by timmyc** [comment]
@youknowriad checking this branch out now.  I agree lets drop the polling for now in this context.  I could potentially see it being useful in a more pure stats context, but on the post listing card, it seems a bit excessive.

File: client/components/data/query-post-stats/index.jsx [hunk 1]
```diff
@@ -15,22 +15,46 @@ class QueryPostStats extends Component {
 	componentWillMount() {
 		const { requestingPostStat, siteId, postId, stat } = this.props;
 		if ( ! requestingPostStat && siteId && postId && stat ) {
-			this.props.requestPostStat( stat, siteId, postId );
+			this.requestPostStat( this.props );
+		}
+	}
+
+	componentWillUnmount() {
+		if ( this.interval ) {
```
Comment (timmyc) [inline client/components/data/query-post-stats/index.jsx]: Could we just call `this.clearInterval()` here and not duplicate the logic?

File: client/my-sites/posts/post-total-views.jsx [hunk 2]
```diff
@@ -87,18 +46,26 @@ var PostTotalViews = React.createClass( {
 		const site = sites.getSite( siteId );
 
 		return (
-			<a href={ '/stats/post/' + postId + '/' + site.slug }
+			<a href={ `/stats/post/${postId}/${site.slug}` }
```
Comment (timmyc) [inline client/my-sites/posts/post-total-views.jsx]: nice 👍

File: client/my-sites/posts/post-total-views.jsx [hunk 3]
```diff
@@ -87,18 +46,26 @@ var PostTotalViews = React.createClass( {
...
 				className={ classNames( {
 					'post__total-views': true,
 					'is-empty': ! viewsCountDisplay
 				} ) }
 				title={ viewsTitle }
 				onClick={ this.props.clickHandler }>
+				<QueryPostStats siteId= { siteId } postId={ postId } stat="views" heartbeat={ HEARTBEAT_IN_SECONDS * 1000 } />
```
Comment (timmyc) [inline client/my-sites/posts/post-total-views.jsx]: Lets do rip out the heartbeat here while we are updating this - but leave the support in `QueryPostStats`

File: client/my-sites/posts/post-total-views.jsx [hunk 4]
```diff
@@ -1,19 +1,20 @@
...
+import StatUpdateIndicator from 'components/stat-update-indicator';
+import Gridicon from 'components/gridicon';
 import sitesList from 'lib/sites-list';
+import QueryPostStats from 'components/data/query-post-stats';
+import { getPostStat } from 'state/stats/posts/selectors';
 
+const	HEARTBEAT_IN_SECONDS = 60;
 const sites = sitesList();
 
 var PostTotalViews = React.createClass( {
```
Comment (timmyc) [inline client/my-sites/posts/post-total-views.jsx]: `viewCount` needs to be added to `propTypes`

**Comment by timmyc** [comment]
blog_posts_ _fly_fishers_place_ _wordpress_com

They aren't the speediest requests either ( though on a slower connection now ).  But more reason to disable the heartbeat for now.

Couple of minor feedback items above but tests out great.  Any PR where the lines of code deleted > lines added is a 🎉 for me!

**Comment by youknowriad** [comment]
I do agree @timmyc the less code we have, the better. We also get a more coherent data approach by dropping all these flux stores step by step.

And I just took your notes into account 😄 tks.

**Comment by timmyc** [comment]
Looks good, thanks for the updates @youknowriad

**Pull request merged**